provider/ansible.py

```diff
@@ -167,16 +172,29 @@ def distro_install(packages="ansible"):
     policy_map = {"distro_install": distro_install,
                   "python_install": python_install}
 
+    ansible_install_policy = params['ansible_install_policy']
+    if (not policy_map.get(ansible_install_policy)
+        or ansible_install_policy not in policy_map):
```

Comment:
Could you explain why it was checked in two different ways ?  We just need check the key in the dict or not , am i right ?

Reply:
You're right, it was a bit "overengineered" ;-) Yes, checking the key in dict is fully sufficient.

Reply:
Done, fixed and tested:

1. Test with `python_install` policy and `ansible-core` previously installed. Installation of Ansible is: OK
2. Test with `python_install` policy and no `ansible` version installed. Installation of Ansible is: OK
3. Test with `distro_install` policy and `ansible-core` previously installed. No `ansible_check_cmd` is defined. Installation of Ansible is: OK
4. Test with `distro_install` policy and no `ansible` version installed. Installation of Ansible is: OK

Note:

- These test results **don't** contain changes from https://github.com/autotest/tp-qemu/pull/3688#discussion_r1206835225.
- I will push them and test when it is confirmed it is correct way to address the issue raised in https://github.com/autotest/tp-qemu/pull/3688#discussion_r1206781905.